Backend/app/email/emailProxy.py

```diff
@@ -0,0 +1,80 @@
+import imaplib
+import email
+
+
+class EmailProxy:
+    imap = None
+
+    def __init__(self, imap_server, email_address, email_password):
+        self.imap_server = imap_server
+        self.email_address = email_address
+        self.email_password = email_password
+
+    def __enter__(self):
+        self.start_connection()
+        return self
+
+    def start_connection(self):
+        """
+        connects to the imap server
+        :return:
+        """
+        self.imap = imaplib.IMAP4_SSL(self.imap_server, port=993)
+        self.imap.login(self.email_address, self.email_password)
+
+        self.imap.select("Inbox")
+        print("connection established")
+        return True
+
+    def spin(self):
+        """
+        fetches emails, calls process_mail if email can be processed
+        :return:
+        """
+        _, msgNums = self.imap.search(None, "UNSEEN")
+
+        for msgNum in msgNums[0].split():
+            _, data = self.imap.fetch(msgNum, "(RFC822)")
+
+            message = email.message_from_bytes(data[0][1])
+
+            if self.can_be_processed(message):
+                print(f"Message Number: {msgNum}")
+                self.process_mail(message)
+        return True
+
+    def can_be_processed(self, email):
+        """
+        should return false if the email was automatically generated or is from blocked user, will be implemented in later sprint
+        :param email:
+        :return boolean:
+        """
+        return True
```

Comment:
Because its not really neccesary right now for the functionality, its just something that would be nice to have in the finished product to avoid possible infinite feedback loops.